source/Engine.cpp

```diff
@@ -2020,19 +2023,36 @@ void Engine::DoCollisions(Projectile &projectile)
 	}
 	else if(projectile.MissileStrength())
 	{
-		// If the projectile did not hit anything, give the anti-missile systems
-		// a chance to shoot it down.
+		// If the projectile did not hit anything, check for anti-missile
+		// systems in range.
 		for(Ship *ship : hasAntiMissile)
 			if(ship == projectile.Target() || gov->IsEnemy(ship->GetGovernment()))
-				if(ship->FireAntiMissile(projectile, visuals))
+				if(ship->IsInAntiMissileRangeOf(projectile))
 				{
-					projectile.Kill();
-					break;
+					antiMissileTargets[ship].push_back(&projectile);
```

Comment:
You can greatly improve the performance of the sort comparator used in `DoAntiMissile` by computing the projectile-ship distance once and storing a `std::pair<Projectile *, double>:
```suggestion
					antiMissileTargets[ship].emplace_back(&projectile, projectile.Position().DistanceSquared(ship.Position()));
```

If you invert the order of the pair, you can use a default-comparator std::set instead of a vector, and get sort-on-insertion due to `std::less`'s behavior comparing `std::pair` by the first and then the second element.

Probably it will be faster to keep the structure as `map<ship *, vector<pair<projectile *, double>>>` and explicitly sort after all insertions, rather than use a collection that sorts on insertion, like `map<ship *, set<pair<double, projectile *>>>`, but one should profile to be certain of that.